contracts/libraries/OracleLibrary.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+
+import '@uniswap/v3-core/contracts/libraries/FullMath.sol';
+import '@uniswap/v3-core/contracts/libraries/TickMath.sol';
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+import '@uniswap/v3-core/contracts/libraries/LowGasSafeMath.sol';
+
+/// @title Oracle library
+/// @notice Provides functions to compute time weighted token conversion rate using the oracle
+library OracleLibrary {
+    uint32 constant UINT32_MAX = uint32(-1);
+
+    /// @notice Calculate time weighted token conversion rate at a time range from a single pool
+    /// @dev This function supports time range that spans across uint32 overflow boundaries
+    /// @param currentBlockTimestamp The current block timestamp
+    /// @param poolAddress The address of the pool to observe
+    /// @param tokenIn The address of the ERC20 token contract of amountIn
+    /// @param amountIn The token input amount before conversion
+    /// @param startBlockTimestamp The start of the time range to be observed
+    /// @param endBlockTimestamp The end of the time range to be observed
+    /// @return amountOut The token output amount after conversion
+    function consult(
+        uint32 currentBlockTimestamp,
+        address poolAddress,
+        address tokenIn,
+        uint256 amountIn,
+        uint32 startBlockTimestamp,
```

Comment:
i would consider taking a single `uint32 period` instead of start/end since that is the common case, and avoids all the edges with it

otherwise, if you want to take start/end, i would just take uint256 timestamps so you can just check that they are ordered, less than current timestamp, and the start timestamp is less than `2**32` seconds ago